### PR TITLE
Added Quick Sort [Python]

### LINKED ISSUE #140: Quick Sort [Python]
[https://en.wikipedia.org/wiki/Quicksort](https://en.wikipedia.org/wiki/Quicksort)

### PR DESCRIPTION
**Pull request by aashutoshrathi**
Fixes #140 

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous timespan of 7 days.

### CODE REVIEW

File: quicksort/quick_sort.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+def swap(x, y):
+    """
+    Perform swap operation between x and y
+    :param x: number to be swapped
+    :param y: number to be swapped with x
+    :return: no returns, swaps values x and y.
+    """
+    temp = x
+    x = y
+    y = x
```
Comment (aviaryan) [inline quicksort/quick_sort.py]: In Python you can do `x,y = y,x`. No need for this function.

Comment (aashutoshrathi) [inline quicksort/quick_sort.py]: removed 👍

File: quicksort/quick_sort.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    Auxiliary Space: O(n)
+    :param array: Iterable of elements
+    :param p: pivot value for array
+    :param r: right limit of array
+    :return: return q value for function, used in partitioning of array.
+    """
+    i = p - 1
+    pivot = array[r]
+    for j in range(p, r):
+        if(array[j] <= pivot):
```
Comment (singhpratyush) [inline quicksort/quick_sort.py]: Redundant parentheses.

File: quicksort/quick_sort.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    """
+    Perform sort using partition function.
+    Time Complexity : O(nlog(n)).
+    Space Complexity : O(n).
+    :param array: Iterable of elements
+    :param pivot: used as left limit of quick sort
+    :param right: right limit for quick sort
+    :return: no returns, sorts array
+    """
+    if(pivot < right):
```
Comment (singhpratyush) [inline quicksort/quick_sort.py]: Redundant parentheses.

Comment (aviaryan) [inline quicksort/quick_sort.py]: `if pivot < right:` would look more pythonic as in Python we can afford to drop brackets in conditions. Same for the other `if` statement also.

**Comment by aashutoshrathi** [comment]
requested changes done 🙇

**Comment by singhpratyush** [comment]
@aviaryan: Are we okay with `i = i + 1` instead of `i += 1`?

**Comment by aviaryan** [comment]
@aashutoshrathi `i += 1` would be better. Thanks for pointing it out @singhpratyush .

**Comment by singhpratyush** [comment]
There are two files in the commit.

File: quicksort/quick_sort.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    :return: return q value for function, used in partitioning of array.
+    """
+    i = p - 1
+    pivot = array[r]
+    for j in range(p, r):
+        if array[j] <= pivot:
+            i += 1
+            array[i], array[j] = array[j], array[i]
+    i += 1
+    array[r], array[i] = array[i], array[r]
```
Comment (mohitkyadav) [inline quicksort/quick_sort.py]: `(array[r], array[i]) = (array[i], array[r])` in 16 and 18th line.

Comment (singhpratyush) [inline quicksort/quick_sort.py]: @mohitkyadav: I don't think that there's anything wrong with this.

Comment (mohitkyadav) [inline quicksort/quick_sort.py]: I was trying to make a comment. Nothing wrong with that btw. Sorry wrong click.

**Pull request merged**